packages/fxa-auth-server/lib/routes/utils/third-party-events.ts

```diff
@@ -31,16 +32,88 @@ export type JWTSETPayload = {
   events: SETEvents;
 };
 
+async function getUidFromSub(sub: string, db: any, provider: Provider) {
+  const { uid } = await db.getLinkedAccount(sub, provider);
+  return uid;
+}
+async function revokeThirdPartySessions(
+  uid: string,
+  provider: Provider,
+  log: any,
+  db: any
+) {
+  const sessions = await db.sessions(uid);
+
+  // Revoke all sessions created from third party logins
+  let deletedCount = 0;
+  for (const session of sessions) {
+    if (session.providerId === PROVIDER[provider]) {
+      await db.deleteSessionToken(session);
```

Comment:
I thought about doing this with only SQL, but opted to use our existing function. This also handles revoking tokens in Redis too. I don't believe there is a signficant impact on performance since this isnt a user facing action.